src/gui/MainWindow.cpp

```diff
@@ -519,6 +519,11 @@ void MainWindow::updateWindowTitle()
         windowTitle = BaseWindowTitle;
     } else {
         windowTitle = QString("%1 - %2").arg(customWindowTitlePart, BaseWindowTitle);
+        if (customWindowTitlePart.right(1) == "*") {
+            setWindowModified(true);
+        } else {
+            setWindowModified(false);
+        }
```

Comment:
@weslly couldn't you add a function similar to `DatabaseTabWidget::readOnly`, used at line 510, to `DatabaseTabWidget`?